cumulus/parachains/common/src/xcm_config.rs

```diff
@@ -78,3 +78,60 @@ impl<Location: Get<MultiLocation>> ContainsPair<MultiAsset, MultiLocation>
 		matches!(asset.id, Concrete(ref id) if id == origin && origin == &Location::get())
 	}
 }
+
+/// Accepts an asset if it is a native asset from the system (Relay Chain or system parachain).
+pub struct ConcreteNativeAssetFromSystem;
+impl ContainsPair<MultiAsset, MultiLocation> for ConcreteNativeAssetFromSystem {
+	fn contains(asset: &MultiAsset, origin: &MultiLocation) -> bool {
+		log::trace!(target: "xcm::contains", "ConcreteNativeAssetFromSystem asset: {:?}, origin: {:?}", asset, origin);
+		let parent = MultiLocation::parent();
+		let is_system = match origin {
+			// The Relay Chain
+			MultiLocation { parents: 1, interior: Here } => true,
```

Comment:
In these `origin: MultiLocation` matches, the `Junction` in this case would be where the next "hop" is from the current location. Also, it seems the current location in this case is the relay chain, is this because these XCM messages are processed on the relay chain or is the starting location always the relay chain?

Reply:
Not sure I understand what you are saying. The origins are relative to the destination of the teleport (in this case the System Parachain). That's why it it `parents: 1` in both cases. It can either come from the Relay Chain or another System Parachain.

Reply:
I was confused about the location where the actual XCM code from the messages is run, now I understand the recipient of the message is supposed to process it so that's the local context.